### PR TITLE
Fix `.signature()` of Inferred Array Types

### LINKED ISSUE #36786: Invalid Signature returned for Inferred Array type in Semantic API
**Description:**
Consider the following inferred array type definition.
```
type MyIntArr1 int[*];
```

The `.signature()` returned for the type descriptor of the above inferred array type definition is `int[]`, where the expected type shall be `int[*]`.

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #36786

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaArrayTypeSymbol.java [hunk 1]
```diff
@@ -36,12 +38,15 @@ public class BallerinaArrayTypeSymbol extends AbstractTypeSymbol implements Arra
     private Integer size;
     private TypeSymbol memberTypeDesc;
     private String signature;
+    private final boolean isInferredArray;
```
[the remark this answers is not in the document]
Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaArrayTypeSymbol.java]: Fixed.

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TypedescriptorTest.java [hunk 2]
```diff
@@ -242,6 +242,27 @@ public Object[][] getArrayVarPos() {
         };
     }
 
+    @Test(dataProvider = "InferredArrayVarPostProvider")
```
Comment (dulajdilshan) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TypedescriptorTest.java:245]:
```suggestion
    @Test(dataProvider = "InferredArrayVarPosProvider")
```

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TypedescriptorTest.java [hunk 3]
```diff
@@ -242,6 +242,27 @@ public Object[][] getArrayVarPos() {
...
+    public void testInferredArrayType(int line, int col, String expSignature) {
+        Symbol symbol = getSymbol(line, col);
+        TypeSymbol type = ((VariableSymbol) symbol).typeDescriptor();
+        assertEquals(type.typeKind(), ARRAY);
+        assertEquals(type.signature(), expSignature);
+    }
+
+    @DataProvider(name = "InferredArrayVarPostProvider")
```
Comment (dulajdilshan) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/TypedescriptorTest.java:253]:
```suggestion
    @DataProvider(name = "InferredArrayVarPosProvider")
```

**Review by dulajdilshan — COMMENTED** [review]
LGTM, let's do these minor changes

**Comment by dulajdilshan** [comment]
Let's update the branch as well

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Comment by sanjana** [comment]
Closing since fixed in #36828

**Pull request closed without merging**